### ISSUE TITLE
500 Error returned when many requests happen back to back

### ISSUE THREAD

**Issue by lukejgaskell** [propublica/congress-api-docs#276]
### Overview
This workflow has worked for me for about the last year, but recently no longer works. It seems to have stopped working around June 16th 2022.

### My workflow (runs once a day)
 - get members of the house: `https://api.propublica.org/congress/v1/117/house/members.json`
 - get members of the senate `https://api.propublica.org/congress/v1/117/senate/members.json`
 - get details for each of those members: `https://api.propublica.org/congress/v1/members/${memberId}.json`
     - launching all the requests as the same time
 
###  Debugging steps
  - I implemented batching and set the size to 25, did not work.
  - Set the batch size to 5, did not even have one successful request.
  - Set the batch size to 2, worked for 8 iterations then failed the requests after that point.
  - Set the batch size to 1, worked for about up to 60ish (inconsistent) requests, then started to fail as well.
  
###   Error Response
  ```
  {
    status: 500,
    statusText: 'Internal Server Error',
    headers: {
      'content-type': 'application/json',
      'content-length': '16',
      connection: 'close',
      date: 'Mon, 27 Jun 2022 16:09:14 GMT',
      'x-amzn-requestid': '27338f5c-3a01-4144-88dd-40cde43ee9ac',
      'x-amzn-errortype': 'AuthorizerConfigurationException',
      'x-amz-apigw-id': 'UY5mrGJmoAMFx_w=',
      'x-cache': 'Error from cloudfront',
      via: '1.1 110142bfecf028552c3361846a29130a.cloudfront.net (CloudFront)',
      'x-amz-cf-pop': 'EWR50-C1',
      'x-amz-cf-id': 'IDXXSw5wyw6dvZZJ9ZMhMuFxtcCKykgV-RHlHIzSoNcIFQdgOszD6A=='
    }
  ```

**Comment by dwillis** [comment]
@lukejgaskell Thanks for this - we recently implemented some additional rate limiting and we've rolled that back. Can you check to see if this still occurs?

**Comment by lukejgaskell** [comment]
Hey @dwillis, thanks for the quick response! That seems to have done the trick. I've adjusted the requests to do batches of 50, but please do let me know if you do plan to change that concurrency limit!

**Comment by dwillis** [comment]
@lukejgaskell sounds good, thanks! would it make sense to offer bulk data on legislators that's updated regularly? If so, how often would make sense?

**Comment by lukejgaskell** [comment]
@dwillis that would be fine too, for my scenario I really only need daily - since I figure legislator data doesn't update that frequently. I'm currently only making the specific member calls for district information, but i'm also querying expenses for all members so I guess it depends on what would come with that bulk data.

**Comment by dawaltconley** [comment]
I ran into this as well for the same use case (getting updated info on MOC no more than once a day). After finding this issue, limiting to 50 concurrent requests seems to have solved it for me as well.

I probably should have guessed that 535 concurrent get requests was a bad idea. 🙃 But I think either documenting the concurrency limits somewhere or clarifying this in the 500 error response message would be helpful.

Besides that, I suppose a way to get more detailed data for all current MOC via a single request could be useful as well. I was trying to [tabulate voting records](https://github.com/dawaltconley/birdog) on certain bills, and I think I just made the extra queries to get their committee assignments.
